### PR TITLE
ignore `redundant_pub_crate` in `useless_attribute`

### PR DESCRIPTION
**Pull request by Alexendoo**
changelog: [`useless_attribute`] no longer lints [`redundant_pub_crate`]

As mentioned in https://github.com/rust-lang/rust-clippy/issues/8732#issuecomment-1106489634

> And it turns out I can't even explicitly allow it at the usage site, because then `clippy::useless_attribute` fires (which would also be a FP?), which is deny-by-default.
> 
> Though it does work if I then allow `clippy::useless_attribute`. 😂
> 
> ```rust
> #[allow(clippy::useless_attribute)]
> #[allow(clippy::redundant_pub_crate)]
> pub(crate) use bit;
> ```
> 
> The originally-reported warning now no longer occurs.

### CODE REVIEW

File: clippy_lints/src/attrs.rs [hunk 1]
```diff
@@ -335,20 +335,19 @@ impl<'tcx> LateLintPass<'tcx> for Attributes {
...
                                             || is_word(lint, sym::deprecated)
                                             || is_word(lint, sym!(unreachable_pub))
                                             || is_word(lint, sym!(unused))
-                                            || extract_clippy_lint(lint)
-                                                .map_or(false, |s| s.as_str() == "wildcard_imports")
-                                            || extract_clippy_lint(lint)
-                                                .map_or(false, |s| s.as_str() == "enum_glob_use")
+                                            || extract_clippy_lint(lint).map_or(false, |s| {
+                                                matches!(
+                                                    s.as_str(),
```
Comment (llogiq) [inline clippy_lints/src/attrs.rs:347]: Why did you change the symbol lookup to a str comparison?

**Review by llogiq — COMMENTED** [review]
This looks good in general, I only have one question.

File: clippy_lints/src/attrs.rs (hunk 1 above)
Comment (Alexendoo) [reply clippy_lints/src/attrs.rs:347]: It was already a str comparison, the change I made was to use `matches!` rather than add a third `|| extract...`

**Comment by llogiq** [comment]
Thank you! @bors r+
